### PR TITLE
Added instructions for enabling/disabling enhancement sync job to enhancements handbook

### PR DESCRIPTION
**Pull request by sreeram-venkitesh**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

Adds instructions to the enhancements handbook about enabling and disabling the test-infra cronjob that syncs KEPs with the `lead-opt-in` label to the enhancements GitHub project board.

#### Which issue(s) this PR fixes:

None. (Didn't create an issue, this was part of a discussion in the enhancements team about the improvements we can make after the code freeze).

#### Special notes for your reviewer:

None.

### CODE REVIEW

**Comment by sreeram-venkitesh** [comment]
Part of https://github.com/kubernetes/sig-release/issues/2375

**Comment by sreeram-venkitesh** [comment]
/remove-lifecycle stale

@salehsedghpour Can you take a look at this? We had discussed these at the end of the 1.29 cycle.

**Comment by salehsedghpour** [comment]
Thanks @sreeram-venkitesh it looks good to me.

When creating the Enhancements Tracking Board for v1.30, I noticed that we may need to give `k8s-infra-ci-robot` the write access. Moreover, according to [this thread](https://kubernetes.slack.com/archives/C2C40FMNF/p1704450942346529) in slack, we may grant `prod-readiness-reviewers` the write access as well.   Am I write @katcosgrove?

May you please consider adding these two around [line 122.](https://github.com/kubernetes/sig-release/pull/2373/files#diff-c853db51181791969e437e9f720a7ef21fbcf460f1eebda80b8ae6a7e2673ba8R122)

File: release-team/role-handbooks/enhancements/README.md [hunk 1]
```diff
@@ -245,6 +246,8 @@ See the sig-architecture Enhancements [KEP Template](https://github.com/kubernet
...
 - If a previously removed Enhancement has had their exception Approved, set their **Enhancement Status** to `tracked for enhancement freeze` in the Enhancement Tracking Board.
 - On Freeze day, email [Kubernetes-Dev](https://groups.google.com/a/kubernetes.io/g/dev) that freeze has happened and upcoming key dates. Examples [1](https://groups.google.com/g/kubernetes-dev/c/JDM7bNKvhqQ/m/8S7BXtXPBQAJ).
+- Disable the [periodic-sync-enhancements-github-project](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-k8s-infra/trusted/sig-release-release-team-jobs/release-team-periodics.yaml#L2) cronjob which syncs enhancements where the `lead-opt-in` label has been added to the GitHub project board.
+  - Disable the job by commenting the `interval` and uncommenting the impossible cron date. You can see how this was done in the v.129 release in [this commit](https://github.com/kubernetes/test-infra/commit/064dd07f0164c2aadc12d611f5a851d6cc40afdd).
```
Comment (npolshakova) [inline release-team/role-handbooks/enhancements/README.md:250]:
```suggestion
  - Disable the job by commenting the `interval` and uncommenting the impossible cron date. You can see how this was done in the v1.29 release in [this commit](https://github.com/kubernetes/test-infra/commit/064dd07f0164c2aadc12d611f5a851d6cc40afdd).
```

**Comment by sreeram-venkitesh** [comment]
@npolshakova Thanks for the review, I've committed the change. 

@salehsedghpour, I've added the instructions for adding the `production-readiness-reviewers` team and `k8s-infra-ci-robot` to the enhancements board in https://github.com/kubernetes/sig-release/pull/2373/commits/bf6dedb993f081022c2de4a9914bbd75a7c367c4. Please take a look!

**Review by npolshakova — APPROVED** [review]
🎉

**Pull request merged**